Azure/Azure-AppServices-Diagnostics-Portal
# Conversational Diagnostics Integration between backend and frontend

## Overview
This PR aims to create a UI component for conversational diagnostics and a service in the UI that facilitates communication with the Conversational Diagnostics backend.

## Type of change


### This PR:
- Adds conversational diagnostics chat component (reusing our existing chat-ui component)
- Adds a conversational diagnostics service that connects to the backend (signalR) over wss.
- Adds a new solutions component for conversational diagnostics that we will modify to fit all our needs in the project.
- Slight UI modifications.

## How Can This Be Tested? <span>&#128269;</span>
- Run the SemanticAppLens Backend project locally.
- Run the portal code locally using npm run ssl-local
- Then go to azure portal with diagnostic extension set to local
- Go to diagnose and solve for a webapp. Click on DiagChat button at the top.
![image](https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/assets/8492235/a7a0ae4d-830f-4a6c-a6a0-4776beb22a65)

- You will see the chat interface
![image](https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/assets/8492235/8f3f2b1b-190b-4caf-9e7f-b4105f64a664)

- Start the chat
![image](https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/assets/8492235/02e2e75e-bca0-4a6f-8f49-563af0177423)

![image](https://github.com/Azure/Azure-AppServices-Diagnostics-Portal/assets/8492235/a4a34e49-dd56-4241-ac0e-41807849cf24)

## Checklist <span>&#9989;</span>
- [x] I have added tests for this feature (if applicable)
- [x] I have incorporated telemetry into this code to capture and analyze relevant metrics (if applicable)
